src/puppetlabs/puppetdb/http/aggregate_event_counts.clj

```diff
@@ -9,25 +10,35 @@
 (defn routes
   [version]
   (app
-   [""]
-   {:get (fn [{:keys [params globals]}]
-           (let [{:strs [query summarize_by counts_filter count_by] :as query-params} params
-                 distinct-options (events-http/validate-distinct-options! query-params)
-                 query-options (-> {:counts_filter (when counts_filter (json/parse-string counts_filter true))
-                                    :count_by count_by}
-                                   (merge distinct-options))]
-             (produce-streaming-body
-              :aggregate-event-counts
-              version
-              query
-              [summarize_by query-options]
-              (:scf-read-db globals)
-              (:url-prefix globals))))}))
+    [""]
+    {:get (fn [{:keys [params globals]}]
+            (if (and (:datasource (:scf-read-db globals))
```

Comment:
Right, I mean in terms of calling it out as a util function somewhere. Giving a name to such a procedure seems valuable.